### PR TITLE
no_std support for reth-revm and fixed tests

### PR DESCRIPTION
**Pull request by martinezjorge**
towards #9478

### CODE REVIEW

File: crates/revm/src/batch.rs [hunk 1]
```diff
@@ -37,6 +40,9 @@ pub struct BlockBatchRecord {
     ///
     /// Empty implies that there is going to be addresses to include in the filter in a future
     /// block. None means there isn't any kind of configuration.
+    #[cfg(not(feature = "std"))]
+    pruning_address_filter: Option<(u64, BTreeSet<Address>)>,
+    #[cfg(feature = "std")]
```
Comment (mattsse) [inline crates/revm/src/batch.rs:45]: ~~I'd prefer if we could use the `hashbrown` crate if we are in no_std.
the hashbrown crate has a hashset type that is no_std~~
we should be able to reuse the Hashset type from revm::precompile instead which is no_std

**Review by mattsse — CHANGES_REQUESTED** [review]
great, one nit

File: crates/revm/src/batch.rs (hunk 1 above)
Comment (martinezjorge) [reply crates/revm/src/batch.rs:45]: Bring in the nitpicks! Will do!

Comment (martinezjorge) [reply crates/revm/src/batch.rs]: If further changes are needed please let me know

**Review by mattsse — APPROVED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -345,7 +345,7 @@ reth-primitives-traits = { path = "crates/primitives-traits", default-features =
 reth-provider = { path = "crates/storage/provider" }
 reth-prune = { path = "crates/prune/prune" }
 reth-prune-types = { path = "crates/prune/types" }
-reth-revm = { path = "crates/revm", default-features = false }
```
Comment (martinezjorge) [inline Cargo.toml]: Why in this case are we removing default-features = false?

Comment (mattsse) [reply Cargo.toml]: this is a workspace wide issue and handling features needs special care, I'm still considering options here on what's the solution. and I want to change this in a special pr

Comment (martinezjorge) [reply Cargo.toml]: thank you for clarifying

**Pull request merged**